src/js/components/CheckBox/CheckBox.js

```diff
@@ -39,7 +40,22 @@ class CheckBox extends Component {
       hidden = <input name={name} type="hidden" value="true" />;
     }
 
-    const Icon = theme.checkBox.icons.checked;
+    if (checked && indeterminate) {
```

Comment:
can we do this in the constructor instead? I don't think it is a good idea to check this in every render function (called multiple times).

Reply:
yes thanks didn't think of that!